src/api.rs

```diff
@@ -57,6 +60,44 @@ impl Api {
         Ok(Json(plan))
     }
 
+    /// Pause a plan
+    #[oai(path = "/plan/pause/:dataset_id", method = "put", tag = Tag::Plan)]
+    async fn plan_pause_put(
```

Comment:
@coderabbitai please create unit test for this endpoint where we set to pause for an unpaused plan.